src/main/scala/za/co/absa/abris/avro/sql/AvroDataToCatalyst.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.abris.avro.sql
+
+import java.security.InvalidParameterException
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.apache.spark.SparkException
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression, UnaryExpression}
+import org.apache.spark.sql.types.{BinaryType, DataType}
+import za.co.absa.abris.avro.format.SparkAvroConversions
+import za.co.absa.abris.avro.parsing.utils.AvroSchemaUtils
+import za.co.absa.abris.avro.read.confluent.{ScalaConfluentKafkaAvroDeserializer, SchemaManager}
+import za.co.absa.abris.avro.serde.{AvroReaderFactory, AvroToRowConverter}
+
+import scala.util.control.NonFatal
+
+case class AvroDataToCatalyst(
+    child: Expression,
+    jsonFormatSchema: Option[String],
+    schemaRegistryConf: Option[Map[String,String]],
+    removeSchemaId: Boolean)
```

Comment:
Maybe `confluentCompliant` or something similar? This way, if Confluent comes up with new stuff we might be able to effect it without changing the API. It is very minor though.